tests/test_edge_cases.py

```diff
@@ -1078,6 +1078,7 @@ class C(A):
         class D(A):
             integer = 'G'
 
+    # TODO: Do we want any changes to this behavior in v2? (Currently, no error is raised)
     assert str(exc_info.value) == (
         'The type of D.integer differs from the new default value; '
         'if you wish to change the type of this field, please use a type annotation'
```

Comment:
Question 1 ☝️

Reply:
I think it should be an error to redefine any field without an annotation - that way we don't need to start trying to infer the type of the default value.